proof/refine/ARM/Reply_R.thy

```diff
@@ -942,6 +942,18 @@ lemma bindScReply_valid_objs'[wp]:
   apply (clarsimp simp: obj_at'_def)
   done
 
+lemma sym_refs_replySCs_of:
+  "\<lbrakk>ko_at' sc scPtr s; scReply sc = Some rptr; sym_refs (state_refs_of' s)\<rbrakk>
+   \<Longrightarrow> replySCs_of s rptr = Some scPtr"
```

Comment:
In my `reply_remove_tcb_corres` PR, there is
```
lemma sym_refs_replySc_scReplies_of:
   "\<lbrakk>reply_at' rp s'; sym_refs (state_refs_of' s'); sc_at' scp s'\<rbrakk>
    \<Longrightarrow> replySCs_of rp = Some scp \<longleftrightarrow> scReplies_of s' scp = Some rp"
 ```
  which might be slightly more general. But we actually don’t need both of `reply_at’` and `sc_at’`.

Reply:
Actually, we don’t need either of them if we have the following:
```
lemma sym_refs_replySc_scReplies_of:
   "\<lbrakk>sym_refs (state_refs_of' s'); pspace_aligned' s’; pspace_distinct’ s'\<rbrakk>
    \<Longrightarrow> replySCs_of rp = Some scp \<longleftrightarrow> scReplies_of s' scp = Some rp"
 ```